src/linalg.h

```diff
@@ -4,12 +4,23 @@
 #include <Eigen/Geometry>
 #include <Eigen/Dense>
 #include <cstdint>
+#include <vector>
 
 using Eigen::Vector2d;
 using Eigen::Vector3d;
 using Eigen::Vector3f;
 using Eigen::Vector3i;
 
+#ifdef _MSC_VER
+  #include <Eigen/StdVector> // https://eigen.tuxfamily.org/dox/group__TopicStlContainers.html
+  #if !EIGEN_HAS_CXX11_CONTAINERS
+    #warning "Eigen has detected no support for CXX11 containers and has redefined std::vector"
+  #endif
+  typedef std::vector<Vector2d, Eigen::aligned_allocator<Vector2d> > VectorOfVector2d;
+#else
+  typedef std::vector<Vector2d> VectorOfVector2d;
+#endif
+
```

Comment:
I ended up fixing Eigen alignment issues the "hard way" before realizing that other builds are just setting `EIGEN_DONT_ALIGN` to workaround these issues.  This section is the main part I'm still not sure about, as I've set it to only trigger if `_MSC_VER` defined.  

This block of code is addressing the issue of "fixed-size vectorizable" eigen objects (Vector2D is multiple of 16bytes) which are used inside of STL containers: https://eigen.tuxfamily.org/dox/group__TopicStlContainers.html
Vector is apparently a special case, where Eigen documentation says to include this `<Eigen/StdVector>` header. Looking at the source, it redefines std::vector if `EIGEN_HAS_CXX11_CONTAINERS` is not defined.  All the builds are using c++11 and should support that as far as I know, so I'm not sure why but it was triggering the warning I placed here(maybe a bug in Eigen's detection).  Consequently Eigen refused to work with initializer lists after redefining `std::vector`, namely here:
https://github.com/openscad/openscad/blob/27ce666e9c9e3f80bd8f022cf94198a816b4901a/src/primitives.cc#L592

Here are some of the build logs with failures just before I wrapped it in the `#ifdef _MSC_VER`:
[Travis](https://travis-ci.org/openscad/openscad/jobs/499237309) 
[CircleCI](https://circleci.com/gh/openscad/openscad/1996?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) 
[AppVeyor](https://ci.appveyor.com/project/kintel/openscad/builds/22687164)

So *maybe* the solution, if we assume Eigen is wrong about whatever `EIGEN_HAS_CXX11_CONTAINERS` is supposed to indicate, is to unconditionally use the `Eigen::aligned allocator` but not the `<Eigen/StdVector>` include?

Anyways the qmake builds on this branch remain unchanged so they still set `EIGEN_DONT_ALIGN` as they have been, to err on the side of caution this close to a new release.  So I think its fine to merge if we want to figure this out in depth later. 

I did briefly test qmake with `EIGEN_DONT_ALIGN` removed in this commit: https://github.com/openscad/openscad/pull/2824/commits/f62d8144ecfcdac224d39ceaf4d3da3415b7c656 
and the builds all passed, (although CircleCI doesn't actually run tests for those builds).
Then reverted that change in most recent commit here: https://github.com/openscad/openscad/pull/2824/commits/35ebcceab82359f096743fc5a38f43ccd98d9342

Reply:
@kintel, @t-paul Any thoughts on this part?

Also adding permalink so the whole section can be viewed:
https://github.com/openscad/openscad/blob/35ebcceab82359f096743fc5a38f43ccd98d9342/src/linalg.h#L14-L22

Reply:
This still scares me a bit, but as long as we don't change behavior for other platforms, we can open a new ticket to review/clean this up later.